### PR TITLE
[DOCS-6112] Removal of Legacy transforms and moving metadata to T-Engines

### PR DESCRIPTION
**Pull request by gravitonian**

### CODE REVIEW

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 1]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
 
-You can test this script from command line before trying the transformation:
-
-```bash
-$ python convertJson2html.py testnode.json test.html
+### Configure a custom Local transform pipeline
+Local Transforms may be combined together in a pipeline to form a new transform, where the output from one becomes the 
+input to the next and so on. A pipeline definition (JSON) defines the sequence of transforms and intermediate Media Types. 
+Like any other transformer, it specifies a list of supported source and target Media Types. If you don't supply any,
+all possible combinations are assumed to be available. The definition may reuse the transformOptions of transformers in the
```
Comment (oliveralfresco2018) [inline content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:168]: transformOptions  needs a space

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 2]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
-![dev-extensions-repo-transformations-json2html-result]({% link content-services/images/dev-extensions-repo-transformations-json2html-result.png %})
-
-You can see that the newly generated HTML file has a thumbnail (refresh the page to see it). If you navigate to the 
-details page a preview will also be available, and you can search for the content in the JSON file and you would see 
-the HTML version of it. You could expand the JavaScript file to move the generated HTML file somewhere else in the 
-repository if you needed to.
-
-The next step is to create a pipeline transformation JSON to HTML to TEXT to make the JSON searchable and to avoid 
-having the extra HTML file generated. This can be achieved with some extra properties in the configuration:
+On startup this location is checked every 10 seconds, but then switches to once an hour if successfully. After a problem,
```
Comment (oliveralfresco2018) [inline content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:213]: successful

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 3]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
-
-You can see that the newly generated HTML file has a thumbnail (refresh the page to see it). If you navigate to the 
-details page a preview will also be available, and you can search for the content in the JSON file and you would see 
-the HTML version of it. You could expand the JavaScript file to move the generated HTML file somewhere else in the 
-repository if you needed to.
-
-The next step is to create a pipeline transformation JSON to HTML to TEXT to make the JSON searchable and to avoid 
-having the extra HTML file generated. This can be achieved with some extra properties in the configuration:
+On startup this location is checked every 10 seconds, but then switches to once an hour if successfully. After a problem, 
+it tries every 10 seconds again. These are the same properties use to decide when to read T-Engine configurations,
```
Comment (oliveralfresco2018) [inline content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:214]: should be used

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 4]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
+* `supportedSourceAndTargetList` - The supported source and target
+Media Types, which refer to the Media Types this failover transformer
+can transform from and to, additionally you can set the priority and the
+maxSourceSizeBytes see [Supported Source and Target List](https://github.com/Alfresco/alfresco-transform-core/blob/master/docs/engine_config.md#supported-source-and-target-list).
+Unlike pipelines, it must not be blank.
+* `transformOptions` - A list of references to options required by
+the pipeline transformer.
+
+### Overriding a Local transform
+In the same way as it is possible combine Local transforms into pipelines, it is also possible to override a
```
Comment (oliveralfresco2018) [inline content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:279]: possible to combine

File: content-services/latest/install/zip/additions.md [hunk 5]
```diff
@@ -4,31 +4,6 @@ title: Install additional software
 
 You can install the third-party software used by Content Services independently. Some of the software can be installed any time before or after installing Content Services.
```
Comment (oliveralfresco2018) [inline content-services/latest/install/zip/additions.md:5]: I guess this wont show? Its not red !

**Review by oliveralfresco2018 — CHANGES_REQUESTED** [review]
Some minor bits but looks good

File: content-services/latest/config/index.md [hunk 6]
```diff
@@ -218,21 +205,25 @@ There are two types of property that you can edit using a JMX interface:
...
+* **IMPORTANT!** Then, the database is checked to see if there are any property values set there, and if any property has been changed, this value is used instead.
 
-Some of the type 2 properties can be viewed and changed by the JMX console, but some can't. For example, `jodconverter.officeHome` can be viewed and changed using the JMX client; `index.recovery.mode` can't be viewed or changed using the JMX client.
+Some of the type 2 properties can be viewed and changed by the JMX console, but some can't. For example, `googledocs.enabled` can be viewed and changed using the JMX client; `index.recovery.mode` can't be viewed or changed using the JMX client.
 
 In a new installation, none of these properties are stored in the database. If you set a property using the JMX interface, Content Services stores the value of the property in the database. If you never use JMX to set the value of a property, you can continue using the `alfresco-global.properties` file to set the value of the property. Once you change the property setting using JMX, and it's stored in the database, and you can't use the properties files to change the value of that property.
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/config/index.md:228]: I don't see any indentation in the code

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 7]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
+By default they are created as primary children of their source item but it is possible to have them created at other places 
+specified explicitly or as templated paths.
 
-The transformer bean definition does not contain the source to target mimetype transformations it supports, rather 
-this is contained in a properties file for easier management by a System Administrator (and these properties can be 
-re-defined in` alfresco-global.properties`). This properties file is also located in the repository JAR 
-(`/alfresco/subsystems/Transformers/default`) and is called `transformers.properties`. The configuration for the 
-`PdfBoxContentTransformer` is as follows:
+The following describes an example with the `reformat` rendering engine and the custom transformation you have previously 
+implemented (that is json to html). You can invoke the rendering engine and create an HTML rendition for a JSON document
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:62]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 8]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
 ```
 
-The `content.transformer.PdfBox.extensions.pdf.txt.maxSourceSizeKBytes=25600` property setting ensures that PDFBox works 
-only with PDF files up to 25MB. Note also that the `content.transformer.TikaAuto.extensions.pdf.txt.maxSourceSizeKBytes=25600` 
-property setting ensures that TikaAuto works only with PDF text files up to 25MB.
-
-The properties have names that follow a certain convention:
+The rendition definition name is a QName that we have to come up with, use a known namespace such as `cm`. 
+The `mime-type` parameter tells the `reformat` rendering engine that the new rendition should be a HTML file. 
+By default this rendition will be store as a hidden child to the uploaded JSON document. If you used the Node Browser
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:75]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 9]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
-to determine what transformations that are used. As the default priority is 100, setting the priority to 50 normally 
-results in the transformer being used. Other compatible transformers will be tried in priority order if the one with 
-highest priority fails for some reason. If multiple transformations are needed to get from source to target mimetype a 
-`pipeline` transformation can be set up. It is also possible to control exactly which transformers are used in case of 
-a failure by using the `failover` property.
+## Transformation changes from Content Services 6.2
+From Content Services version 6.2 it is possible to create custom transforms that run in separate processes known as 
+T-Engines (short for Transformer Engines). The same engines may be used in Community and Enterprise Editions. They may be 
+directly connected to the Repository as Local Transforms, but in the Enterprise edition there is the option to include 
+them as part of the Alfresco Transform Service (ATS) which provides more balanced throughput.
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:90]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 10]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
-If you are running an Enterprise edition these properties may be changed via JMX while Content Services is 
-running (note, any changes via JMX and database takes precedence over any property file settings).
+A T-Engine is intended to be run as a Docker image, but may also be run as a standalone process.
 
-You can create custom content transformers to transform one type of content into another, where that transformation 
-is not already supported, such as when you have a custom input content type, or custom output content type. Take the 
-JSON mimetype for example, if you upload a JSON file to the repository it will not have a thumbnail or a preview, and 
-it will not be indexed and searchable.
+Prior to Content Services 6.0 Legacy transformers ran within the same JVM as the Repository. They and their supporting classes were 
+deprecated and in Content Services 6, but have now been removed. Content Services 6.2 still uses them if a rendition cannot be created by the
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:95]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 11]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
- 2016-01-05 07:50:15,117  DEBUG [content.transform.TransformerDebug] [http-bio-8443-exec-9] 10  Finished in 7 ms Transformer NOT called
+localTransform.<engineName>.url=
 ```
 
-You can also see that there is no transformations available for the JSON mimetype on the 
-`http://localhost:8080/alfresco/service/mimetypes?mimetype=application/json#application/json` admin page:
+The `<engineName>` is a unique name of the T-Engine. For example, `localTransform.helloworld.url`. Typically a T-Engine
+contains a single transform or an associated group of transforms. Having set the URL to a T-Engine, the repository will 
+update its configuration by requesting the [T-Engine configuration](https://github.com/Alfresco/acs-packaging/blob/master/docs/creating-a-t-engine.md#t-engine-configuration){:target="_blank"} 
+on a periodic basis. It is requested more frequent on start up or if a communication or configuration problem has
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:115]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 12]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
+Local or Transform Service transforms can be enabled or disabled independently of each other. The repository will try to 
+transform content using the Transform Service if possible and falling back to a Local Transform. If you are using Share, 
+Local Transforms are required, as they support both synchronous and asynchronous requests. Share makes use of both, so
+functionality such as preview will be unavailable if Local transforms are disabled. The Transform service only supports 
+asynchronous requests.
 
-Currently the transformer would halt on the Python script call as that has not yet been implemented. Download the 
-`json2html` python module as follows:
+The following sections will show how to create a Local transform pipeline, a Local transform failover or a Local transform 
+override, but remember that they will not be used if the Transform Service (ATS) if it is able to do the transform and
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:148]: Fixed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md (hunk 1 above)
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:168]: changed to `transformOptions`

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md (hunk 2 above)
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:213]: FIxed

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md (hunk 4 above)
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:279]: Fixed

File: content-services/latest/install/zip/additions.md (hunk 5 above)
Comment (gravitonian) [reply content-services/latest/install/zip/additions.md:5]: Haven't touched anything in zip installs...

File: content-services/latest/develop/repo-ext-points/content-transformers-renditions.md [hunk 13]
```diff
@@ -43,431 +42,502 @@ Content Services. So when accessing the "mimetypes" Web Script make sure the mim
...
-JSON mimetype for example, if you upload a JSON file to the repository it will not have a thumbnail or a preview, and 
-it will not be indexed and searchable.
+Prior to Content Services 6.0 Legacy transformers ran within the same JVM as the Repository. They and their supporting classes were 
+deprecated and in Content Services 6, but have now been removed. Content Services 6.2 still uses them if a rendition cannot be created by the 
+Transform Service or Local Transforms. The process of migrating custom legacy transformers is described at the end
+of this page.
 
-The following will show up in the logs when debugging is turned on (turn it on via 
-`log4j.logger.org.alfresco.repo.content.transform.TransformerDebug=DEBUG and log4j.logger.org.alfresco.repo.content.transform=DEBUG`):
+One of the advantages of Custom Transforms and Renditions in Content Services 6.2 is that there is no longer any need for custom
```
[the remark this answers is not in the document]
Comment (gravitonian) [reply content-services/latest/develop/repo-ext-points/content-transformers-renditions.md:99]: Changed to 6.2 and above

**Pull request merged**